drivers/rtc: Enable RTC driver for STM32F3/F4 series

ydamigos opened the pull request:
Enable RTC driver for STM32 F3 and F4 series.
Tested on stm32f3_disco, 96b_carbon and nucleo_f401re.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos committed:
drivers/rtc: Enable RTC driver for STM32F4 series

Enable RTC driver for STM32F4 series.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos committed:
dts/arm/st: Add rtc node to STM32F4

Add rtc node to STM32F4 series.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos committed:
boards/arm: Enable RTC on STM32F4 based boards

Enable RTC on STM32F4 based boards.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>
Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ydamigos commented:
@kzyapkov Could you also check it?

erwango commented on arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.series:
```diff
@@ -32,4 +32,11 @@ config I2C_STM32_V1
 
 endif # I2C_STM32
 
+if RTC
```
Since it will likely be common for all series, why not moving this in 
arch/arm/soc/st_stm32/common/Kconfig.defconfig.series ?

ydamigos replied:
Done.

ydamigos committed:
drivers/rtc: Enable RTC driver for STM32F3 series

Enable RTC driver for STM32F3 series.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos committed:
dts/arm/st: Add rtc node to STM32F3

Add rtc node to STM32F3 series.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>
Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ydamigos committed:
boards/arm: Enable RTC on STM32F3 based boards

Enable RTC on STM32F3 based boards.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>
Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

kzyapkov commented:
Looks good to me, works on F401RE.

jli157 commented on arch/arm/soc/st_stm32/common/Kconfig.defconfig.series:
```diff
@@ -72,4 +72,11 @@ config USB_DC_STM32
 
+if RTC
+
+config RTC_STM32
```
The new patch doesn't support STM32F1 series, right? Then, it should not be default y for all STM32 series.

ydamigos replied:
You are right. The driver doesn't supoort STM32F1 series. I will update the PR.

erwango replied:
Well, I think it makes sense to put it in common.

RTC should not be enabled if not implemented for one series (F1 in the case you mention).
RTC_STM32 should be used if RTC is enabled, this is true for all STM32 series (would not be the case if 2 versions of STM32 RTC driver were available as a choice to the user).

This is not different from any other driver in arch/arm/soc/st_stm32/common/Kconfig.defconfig.series

ydamigos replied:
```
if RTC && !SOC_SERIES_STM32F1X
        def_bool y
endif # RTC && !SOC_SERIES_STM32F1X
```
@erwango Does the above works for you? It makes clear that the RTC_STM32 driver doesn't support STM32F1 right now.

erwango replied:
But then what for other series? Is there something different for RTC for STM32F1 different vs RTC driver for other series not supported yet (F0/F2/F7/L0)?
Besides, use of negative is complex to maintain and should be updated each time a new series is added.

@jli157 : To me, current use and role of `common/Kconfig.defconfig.series` is to enable STM32 versions of drivers not to do any gating.
If one enables RTC on F1, he'll face issues because it is not available yet. Such as the one enabling SPI for a recently introduced series.

ydamigos replied:
@erwango I get you point, I have updated the PR.

erwango replied:
Thanks. Completing the review right now

ydamigos commented:
recheck

galak reviewed (approved):
Why do the rtc nodes have #addr & #cell sizes properties

galak commented on drivers/rtc/Kconfig.stm32_rtc:
```diff
@@ -15,7 +15,7 @@ config RTC_STM32
 	select USE_STM32_LL_EXTI
 	select NEWLIB_LIBC
```
While not specific to your change, but does the RTC driver really need NEWLIB?

ydamigos replied:
It helps with the conversion between calendar daytime and UNIX timestamp.

galak committed:
dts/arm/st: Remove unneeded setting of #{address,size}-cells from RTC

The RTC node for some reason was setting #address-cells and #size-cells.
However it has no children so there isn't really any reason to do this.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ydamigos commented:
recheck
